### DISCUSSION TITLE
Dexcom Stelo - xdrip support

### DISCUSSION THREAD

**Discussion by horacioofman** [NightscoutFoundation/xDrip#3645] [Q&A] [answered]
Hi,
Dexcom Stelo cgm is now available on their website for $90 per month.

Does anybody know if xdrip currently supports Stelo?
I will be buying a unit to give it a try and will share my experience, but If someone has already tried, please let me know if you were able to connect xdrip to stelo and   if any special settings were needed to make it work.

Thanks

**Comment by Navid200** [comment]
Can you post a screenshot?
Do you see the app in the play store?

**Reply by horacioofman** [reply]
HI Navid,
Screenshot? Not sure what you mean. I am asking if anybody has ever connected a Dexcom Stelo (new product) to the Xdrip app.
Check www.dexcom.com or www.stelo.com

which app do you want me to check?  yes, Stelo has an app in the play store, but only works in the USA.

**Reply by Navid200** [reply]
Bring up the app and take a screenshot showing the URL in the browser and post here.

**Comment by gui-dos** [comment]
https://apkpure.com/stelo-by-dexcom/com.dexcom.stelo

**Reply by Navid200** [reply]
Thanks

**Comment by EasternPA** [comment]
Someone confirmed that the Stelo uses DX followed by two numbers and two letters as the Bluetooth device name. I see xDrip+ already looks for DXCM (the G7) and DX02. I don't know if 02 is the G6, but it matches the "two numbers" that the Stelo user confirmed.

Edit: Just got DX01xx as the first confirmed Stelo Bluetooth device name.

**Reply by Navid200** [reply]
One of my concerns with enabling Stelo on xDrip is the possibility of someone who has type 1 diabetes using it for cost saving.  

Dexcom performed research to find out how fast human blood glucose can vary.  From that, they used Nyquist rate to confirm that if there is one measurement every 5 minutes, the variations can be reasonably accurately captured.  

Stelo gives one reading every 15 minutes.  If you use insulin and you see you are high and you take more insulin to correct, by the time you get the next reading, you may already have started to drop a lot.  This can lead to a serious problem.

Dexcom has got approval to market Stelo specifically for type 2 diabetics who do not use insulin.
So, that is my worry and concern.

**Reply by EasternPA** [reply]
I understand your concern. Of course someone could fork it to get it working, but I hope the device will not be banned in the base code. I could see disabling support for insulin in the base app when connected to a Stelo. I prefer xD for privacy reasons, and would like to see support for the Stelo.

**Reply by Navid200** [reply]
Send me an email.

**Reply by dakotaKat** [reply]
For what it's worth -- according to the 510k, the Stelo sensor is still capturing every 5 minutes. I'd also like to use xD with a Stelo for data privacy and portability concerns.

https://www.accessdata.fda.gov/cdrh_docs/pdf23/K234070.pdf

> The proposed Stelo System is based on the same mode of operation and mechanism of reaction as the
predicate G7 CGM System (K231081), which uses a wire type sensing mechanism that continuously measures
interstitial fluid glucose levels and a BLE enabled radio transmitter to wirelessly communicate CGM data to
compatible display devices at regular 5-minute intervals. These data are also able to be reliably and securely
transmitted to other digitally connected devices, excluding insulin devices such as insulin pens and Automated
Insulin Dosing (AID) systems.
The Stelo System uses the same hardware design as the predicate G7 CGM System. The Stelo GSS firmware
is designed to support a factory-calibrated only device (without calibration inputs), to extend the sensor wear
duration from 10 to 15 days while maintaining the accuracy of the device, and to connect to authorized display
devices only (i.e., Stelo MAS). The Stelo MAS includes a redesigned user interface (UI) tailored to the Stelo
System’s user population to simplify the use of the device.

**Reply by EasternPA** [reply]
"authorized display devices only" is disconcerting.

**Reply by jbr7rr** [reply]
> Stelo gives one reading every 15 minutes. If you use insulin and you see you are high and you take more insulin to correct, by the time you get the next reading, you may already have started to drop a lot. This can lead to a serious problem.

From the manual:
"The Stelo Glucose Biosensor System records glucose readings (sensor readings) every 5 minutes. All the sensor readings are then displayed on the mobile app every 15 minutes" 

Tim also made a good comparison about with the G7:
https://www.diabettech.com/cgm/dexcoms-stelo-by-numbers-could-a-15-day-g7-be-imminent/?utm_source=dlvr.it&utm_medium=facebook&utm_campaign=dexcoms-stelo-by-numbers-could-a-15-day-g7-be-imminent

Question then is, does the Stelo send data every 5 min or every 15 minutes, if it is the latter it would be useless in loop settings and also for alarms

**Reply by antonywu** [reply]
> Dexcom has got approval to market Stelo specifically for type 2 diabetics who do not use insulin.
> So, that is my worry and concern.

If user chooses to ignore manufacturer's warnings and recommendations, that is dangerous but I don't think it is up xD+ to decide for the user. At the current rate, my Stelo data is connected to Google Health Connect... would really like to get the data out without reinventing the wheel solely because xD+ wants to protect users from making the wrong choice.

**Reply by horacioofman** [reply]
@dakotaKat @jbr7rr 
I confirm that Stelo is recording every 5 minutes, but the transmissions of such readings it is done on 15 minutes intervals. I timed a cycle and it was closer to 18 minutes really in between 2 transmissions.

> Question then is, does the Stelo send data every 5 min or every 15 minutes, if it is the latter it would be useless in loop settings and also for alarms

T2 diabetics and other Stelo users are not really that concerned in alarms. The 5 records makes for a much nicer and useful curve. I am guessing that transmissions every 15 minutes helps saving battery to extend the sensor life to 15 days.

**Comment by jbfeutz** [comment]
I have one on order and plan to test when it arrives...

I agree that caution should be urged when using Stelo for folks on insulin. Even if today it is identical to the G7, Dexcom is not under the same FDA scrutiny to maintain that. They could elect to perform cost outs on the product that would keep it suitable for use by non-insulin users but be dangerous for insulin users. There would not necessarily be notification of that change.

**Comment by horacioofman** [comment]
My stelo arrived yesterday and I have one week left on my current g7 sensor.
I plan to test them on xdrip as soon as my current sensor expires next week. The packaging and the devices look almost identical. I would be surprised if they are not compatible.

Note: I am a type II diabetic, not taking any insulin. The reason I  use xdrip and I am interested in connecting with thye stelo device is because the Dexcom apps are not available in my country. I've been using xdrip for a couple years now first with the Libre I sensors and lately with the G7. Neither Libre or Dexcom apps are available on my country, and I am buying the devices in a diffent country and then shipping them online.

**Comment by KunaniGaming** [comment]
I have Stelo. I'm on my second sensor now and found Xdrip by googling around trying to find a more realtime way to get my readings into other apps correctly. The stelo app sends data to Health Connect but its 3 hours behind. Also the time/date data must be incompatible with Samsung Health because SH will import and tag the data as current time but the data was from 3 hours ago on Stelo (correct time). This is frustrating when using other apps to do food logging and show glucose data in a chart. I found nightscount which let me here. 

Does xDrip work with the Stelo sensor? As anyone successfully connected and used it yet? Also, if I connect xDrip will that interfere with the Stelo app communicating with the sensor? 

I'm also an app developer so if I can't find a solid solution I'll need to dig deeper to possibly make one.

**Reply by knomura45242** [reply]
I'm wearing stelo now , second day. But I'm struggling to see the data on xdrip. Is there anyhow to read and manage stelo data? Since Stelo app. Does not have any features of calibration,  sharing data, widget, smartwatch reading include G-watch app. So I'm hoping to read data through xdrip+.
Since my medicare insurance rejected supplying G7 due to strict apply of FDA guidline I have to change to Stelo.

**Reply by kaz45242** [reply]
Yes, I love to accept your support on this matter.  Stelo app is not fullfilling my expectations such as caliblation, sharing others, smartwatch reading, and eventlog onto glucose chart.  I've been enjoyed these features with xdrip+/G7, but not now. What kind of information should  i share?  Please let me know.

**Comment by rplakas** [comment]
I am on my second Stelo sensor and they do have the DX##xx for the Bluetooth. Haven't been able to get Xdrip+ to see the Stelo senor though. Anyone else get it connected?

**Reply by Navid200** [reply]
Send me an email

xDrip dot Navid at gmail dot com

**Comment by Hotsolce** [comment, marked as answer]
It looks like xdrip+ now has support for Stelo unless I'm mistaken I think this can be marked as Answered?

[Docs reference](https://navid200.github.io/xDrip/docs/Dexcom/G7.html)

**Comment by davidacm** [comment]
I want to be sure that Stelo is compatible with xdrip. When it asks me to select the data source, stelo does not appear as an option.
I don't know if selecting the "g4, g5, g6, g7, 1" option would work?
The Stelo app is not available in my country. Before purchasing a Stelo, I want to be sure that I'll have an application with which to read the data.
I expose myself to many inconveniences, such as xdrip not being accessible with Talkback, since I'm visually impaired. So I prefer to minimize at least the incompatibility risks.

I don't have any health risk situation, I just want to make sure that everything is fine and how meals influence it. I think Stelo is ideal for these cases.

If Stelo is already supported, it should appear as a supported device when choosing the source, and in the application documentation. Many of us could benefit from this.

The version I have installed is the following release:
https://github.com/NightscoutFoundation/xDrip/releases/download/2024.11.26/xDrip-plus-20241126-a2bbca9.apk

Best regards,
David.

**Comment by horacioofman** [comment]
@davidacm 
Stelo is supported on the latest xdrip version.
See https://navid200.github.io/xDrip/docs/Dexcom/G7.html for setup guide

**Comment by knomura45242** [comment]
Hello, I'm using xdrip+ & stelo. It works well, but a couple of problem
exist. One is calibration does not work. And second is  that my watch,
samsong watch 7 does not connect to stelo directly.  My watch can get data
and input data through my phone, Samsung s24 ultra, but cannot connect to
the sensor with Bluetooth. Otherwise I'm using them without any problems.
I'm satisfying with these setup.  If you have any question, feel free to
contact.

Kaz
